src/LanguageServer/Impl/Implementation/CompletionAnalysis.cs

```diff
@@ -884,11 +881,17 @@ private IEnumerable<CompletionItem> GetModules()
         private string MakeOverrideCompletionString(string indentation, IOverloadResult result, string className) {
             var sb = new StringBuilder();
 
-            sb.AppendLine(result.Name + "(" + string.Join(", ", result.Parameters.Select(MakeOverrideDefParamater)) + "):");
+            var self = result.SelfParameter != null ? new[] { result.SelfParameter } : Array.Empty<ParameterResult>();
+            var parameters = self.Concat(result.Parameters).ToArray();
+
+            sb.AppendLine(result.Name + "(" + string.Join(", ", parameters.Select(MakeOverrideDefParamater)) + "):");
             sb.Append(indentation);
 
-            if (result.Parameters.Length > 0) {
-                var parameterString = string.Join(", ", result.Parameters.Skip(1).Select(MakeOverrideCallParameter));
+            if (parameters.Length > 0) {
+                var parameterString = string.Join(", ", 
+                    result.Parameters
+                        .Where(p => p.Name != "self")
```

Comment:
I'm not sure this is correct. There's nothing special about the name `self` other than that it's the first parameter. It's not a reserved name like `this` would be in something like C# or C++. I think this change would also break `@classmethod` overrides, potentially.

Reply:
This is hacky way to improve another hack. Problem is that code used some acrobatics with `internal _fromFunction` to control if properly returning parameters with overloads should skip first argument. I am not into reworking this since it is all DDG specific. The only place where is seems to matter is here where snippets for overrides are generated in completion. If there is an issue with class methods here, we should add a test and fix the case here.

Reply:
I'm just trying to say that this:

```python
class A(object):
    def foo(x, y):
        x.y = y

    @classmethod
    def foo(p, q):
		p.q = q
```

Should be just as well supported as:

```python
class A(object):
    def foo(self, y):
        self.y = y

    @classmethod
    def foo(cls, q):
		cls.q = q
```

When generating overrides in class derived from `A`. The old code would just skip the first thing, which should be correct regardless of it being a classmethod or not.

I guess one question I have is that it seems to me like this function is only ever called when the completion item is being generated in a class, in which case an override should only show up in a place where the first argument is the thing to skip. Is that not actually the case here? Maybe I'm just not understanding what you're saying the `fromFunction` thing affects.

Reply:
Corrected